src/common/graphql.js

```diff
@@ -1,43 +1,79 @@
 import { ApolloClient, InMemoryCache, gql } from '@apollo/client'
+import { utils } from 'ethers'
 import defaults from './defaults'
 
-const client = new ApolloClient({
-	uri: defaults.api.graphUrl,
-	cache: new InMemoryCache(),
-})
-
-const getXVaderPriceByBlock = async (block) => {
-	const tokensQuery = block ? `
-		query {
-			globals(block: { number: ${block} }) {
-				id
-				name
-				value
-			}
+const getXVaderPrice = async (type = 'Hour', skip = 0, first = 1) => {
+	const query = skip > 0 ? `
+	{
+		globals(
+			first: ${first}
+			skip: ${skip},
+			orderBy: timestamp,
+			orderDirection: desc,
+			where:{ 
+				name: "XVADER_PRICE",
+				type: "${type}"
+			}) {
+			id
+			name
+			value
+			type
+			timestamp
 		}
-	` : `
-		query {
-			globals {
-				id
-				name
-				value
-			}
+	}
+	` :	`
+	{
+		global(id: "XVADER_PRICE") {
+			id
+			name
+			value
 		}
+	}
 	`
 	try {
-		const result = await client.query({ query: gql(tokensQuery) })
-		if (!result || !result.data || !result.data.globals) {
-			return null
+		const result = await queryGraphQL(query, defaults.api.graphql.uri.vaderProtocol)
+		if (result) {
+			return result
+		}
+	}
+	catch (err) {
+		console.log(err)
+	}
+}
+
+const getXVaderApr = async (type) => {
+	const [currentPrice, daysAgoPrice] = await Promise.all([
```

Comment:
i think should change variable name, something like previousPrice. It can be hour price now.

Reply:
True I lost that somewhere